### DISCUSSION TITLE
When to use `nuqs/server`?

### DISCUSSION THREAD

**Discussion by FezVrasta** [47ng/nuqs#717] [Q&A] [answered]
If I wanted to write isomorphic code (working on both server and client) should I import `parseAsString` (for example) from the `nuqs/server` or from somewhere else? Since the last update I can't import from `nuqs` anymore because it's marked as `use client`.

**Comment by franky47** [comment]
Yes, you can import the parsers from `nuqs/server`, which doesn't contain the `"use client"` directive.

**Reply by FezVrasta** [reply]
Thanks for the quick reply.

Do you think we could have a different organization? Maybe `nuqs/` exporting just environment agnostic functions, and `nuqs/client` and `nuqs/server` for the respective environments? Right now it's a bit confusing.

Or maybe just a `nuqs/parsers` export.

**Reply by franky47** [reply]
The `nuqs/parsers` export was there in v1, but I renamed it to `nuqs/server` as more server-centric features were exported (the cache & serializer).

I'll try and make the docs more clear about this.

**Reply by guptaashwanee** [reply]
Do you have any plans to make nuqs/server usable with any other backend framework to parse the query string, as nuqs for front-end (react - vite)

**Reply by franky47** [reply]
@guptaashwanee most of `nuqs/server` is framework agnostic, the only thing that depends on React is the cache feature. If we were to move that to a different import (which would be a breaking change, so not until v3), then you'd be able to use `nuqs/server` to parse/serialize search params anywhere (a nice API to do this from an object describing your search params should land soon, as it will be used in Remix / React Router).

Until then, you'd need to add React canary to your backend project for the `import { cache } from 'react'` to resolve.

**Reply by franky47** [reply]
FYI, there is a temporary solution coming in #805 to allow importing server-side features without bringing along the cache (and a dependency on React):

```ts
import { parseAsXYZ } from 'nuqs/server/parsers'
import { createSerializer } from 'nuqs/server/serializer'
```

The plan is to remove the cache feature from the main `nuqs/server` import in v3 (it will be accessible at `nuqs/server/cache`). Those temporary imports per-feature may also be removed to keep `nuqs/server` the main import.

**Reply by guptaashwanee** [reply]
This would be very great feature.

**Comment by franky47** [comment, marked as answer]
Note: nuqs@2.4.1 shipped with a fix for this in #923. It allows importing from `nuqs/server` without getting the "Cannot find export `cache` from React" error.

Note that you would only need to import from `nuqs/server` in Next.js when using the app router, in code that can be shared between server and client.

For all other cases, importing from `nuqs` is fine (other frameworks don't care about the "use client" directive).
